build.sh

```diff
@@ -18,34 +18,159 @@ EOF
 WARNING: Neither \`libtoolize' nor \`glibtoolize' have been found!
     Please make sure that one of them is installed and is in one of the
     directories listed in the PATH environment variable.
 EOF
-        GLOBAL_ERROR_INDICATOR=1
+            GLOBAL_ERROR_INDICATOR=1
+        fi
     fi
- fi
 
-if test "$GLOBAL_ERROR_INDICATOR" != "0"; then
-    exit 1
-fi
+    if test "$GLOBAL_ERROR_INDICATOR" != "0"; then
+        exit 1
+    fi
+}
+
+build()
+{
+    echo "Building..."
+    check_for_application lex bison autoheader aclocal automake autoconf pkg-config
+    setup_libtool
+
+    set -x
+    autoheader \
+    && aclocal -I m4 \
+    && $libtoolize --copy --force \
+    && automake --add-missing --copy \
+    && autoconf
+}
+
+build_cygwin()
+{
+    echo "Building for Cygwin..."
+    check_for_application aclocal autoconf autoheader automake bison flex git make pkg-config x86_64-w64-mingw32-gcc
+    setup_libtool
+
+    set -e
+
+    : ${INSTALL_DIR:="C:/PROGRA~1/collectd"}
+    : ${LIBDIR:="${INSTALL_DIR}"}
+    : ${BINDIR:="${INSTALL_DIR}"}
+    : ${SBINDIR:="${INSTALL_DIR}"}
+    : ${SYSCONFDIR:="${INSTALL_DIR}"}
+    : ${LOCALSTATEDIR:="${INSTALL_DIR}"}
+    : ${DATAROOTDIR:="${INSTALL_DIR}"}
+    : ${DATADIR:="${INSTALL_DIR}"}
+
+    echo "Installing collectd to ${INSTALL_DIR}."
+    TOP_SRCDIR=$(pwd)
+    MINGW_ROOT="/usr/x86_64-w64-mingw32/sys-root/mingw"
+    GNULIB_DIR="${TOP_SRCDIR}/_build_aux/_gnulib/gllib"
 
-set -x
+    export CC="x86_64-w64-mingw32-gcc"
+
+    mkdir -p _build_aux
+
+    cd _build_aux
+    if [ -d "_gnulib" ]; then
+        echo "Assuming that gnulib is already built, because _gnulib exists."
+    else
+        git clone git://git.savannah.gnu.org/gnulib.git
+        cd gnulib
+        git checkout 2f8140bc8ce5501e31dcc665b42b5df64f84c20c
+        ./gnulib-tool \
+          --create-testdir \
+          --source-base=lib \
+          --dir=${TOP_SRCDIR}/_build_aux/_gnulib \
+          canonicalize-lgpl \
+          regex \
+          sys_socket \
+          nanosleep \
+          netdb \
+          net_if \
+          sendto \
+          gettimeofday \
+          getsockopt \
+          time_r \
+          sys_stat \
+          fcntl-h \
+          sys_resource \
+          sys_wait \
+          setlocale \
+          strtok_r \
+          poll \
+          recv \
+          net_if
+
+        cd ${TOP_SRCDIR}/_build_aux/_gnulib
+        ./configure --host="mingw32" LIBS="-lws2_32 -lpthread"
+        make 
+        cd gllib
+
+        # We have to rebuild libgnu.a to get the list of *.o files to build a dll later
+        rm libgnu.a
+        OBJECT_LIST=`make V=1 | grep "ar" | cut -d' ' -f4-`
+        $CC -shared -o libgnu.dll $OBJECT_LIST -lws2_32 -lpthread
+        rm libgnu.a # get rid of it, to use libgnu.dll
+    fi
+    cd "${TOP_SRCDIR}"
+
+    set -x
+    autoheader \
+    && aclocal -I /usr/share/aclocal \
+    && $libtoolize --copy --force \
+    && automake --add-missing --copy \
+    && autoconf
```

Comment:
is there no autoreconf on cygwin that can replace the above 5 lines?

Reply:
I was following how it builds it for other platforms, but autoreconf seems to work just as well and is simpler :)

Reply:
Ah indeed! We should probably just do that for all platforms then. Care to do a separate PR for that?